pydicom/dataset.py

```diff
@@ -1089,14 +1089,30 @@ def clear(self):
         """Delete all data elements."""
         self._dict.clear()
 
-    def pop(self, *args, **kwargs):
-        return self._dict.pop(*args, **kwargs)
+    @staticmethod
+    def _handle_tuple_and_keyword(fct, key, *args):
+        if isinstance(key, (str, compat.text_type)):
+            tag = tag_for_keyword(key)
```

Comment:
Is checking for string necessary given the new Tag instantiation using keyword in #829?  Could this just go straight to try Tag(key)?

Reply:
Ah, right - I forgot about that! Will change this tonight, thanks.

Reply:
Ok, done.